bool request
---
int64 stamp
bool left_open
bool right_open